packages/react/src/components/ContainedList/ContainedListItem/ContainedListItem.tsx

```diff
@@ -96,7 +105,7 @@ ContainedListItem.propTypes = {
   /**
    * Provide an optional icon to render in front of the item's content.
    */
-  renderIcon: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+  renderIcon: PropTypes.func,
```

Comment:
We can't narrow this, it would be a breaking change for those using an object with this prop.

Reply:
will change this.